### ISSUE TITLE
causal_forest: how do the "sample.fraction"  and "honesty.fraction" options work and differ? 

### ISSUE THREAD

**Issue by sudonghua91** [grf-labs/grf#1027]
Hello, 

My question is as follows: 

I want to build a causal_forest and make out-of-bag predictions. I am confused about the sample.fraction. In causal_forest function, I have used V (training sample) but why sample.fraction is still there as an argument? Does it mean that it further cut the training sample to for example half? If that is the case, it seems more confusing if honesty. fraction cuts it one more time!  
What is the real train sample now? Should I put the whole dataset ("data") in the function causal_forest instead s.t. sample.fraction can split the whole dataset into training sample and test sample once and honesty.fraction further cuts the training sample into halfs? 

One more thing: what does the argument "newdata" mean? Can I use the trained forest to predict the full sample? i.e. the covariates of the full sample (X) are plugged in as the "newdata" in predict function. 
Thank you if anyone can help.

Codes: 

sample = sample.split(data$assignedtutee, SplitRatio = .4)
train = subset(data, sample == TRUE)
test  = subset(data, sample == FALSE)
test = test[,c("female","z_math_rd1","bothabsent_rd1","hometutortime_rd1","avg_class_math_rd1")]
num.trees <- 10000

V = train[,c("female","z_math_rd1","bothabsent_rd1","hometutortime_rd1","avg_class_math_rd1")]
Y = train[,c("delta_z_math")]
W = train[,c("assignedtutee")]
P = train[,c("weight")]

forest<-causal_forest(V, Y, W,
                      num.trees = 10000,
                      sample.weights = P,
                      sample.fraction = 0.5,
                      min.node.size = 3)
predictions = predict(forest, test) # this means that use the trained forest to predict "test" subsample, right? #
predictions = predict(forest) # this means that make out-of-bag predictions on train sample, right? #
X=data[,c("female","z_math_rd1","oneabsent_rd1","bothabsent_rd1","hometutored_rd1","hometutortime_rd1","avg_class_math_rd1")]
predictions = predict(forest, X) # cause overfitting? #

**Comment by erikcs** [comment]
Hi @sudonghua91, the [GRF Reference](https://grf-labs.github.io/grf/REFERENCE.html#samplefraction) has a section on sample fraction and honesty:

>The sample.fraction parameter is a number in the range (0, 1] that controls the fraction of examples that should be used in growing each tree. By default, sample.fraction is set to 0.5. As noted in the section on honest forests, the fractional subsample will be further split into halves when honesty is enabled.

You can find more details on honesty here: https://grf-labs.github.io/grf/REFERENCE.html#honesty-honestyfraction-honestypruneleaves

For predict, newdata is an optional test matrix to predict at, see https://grf-labs.github.io/grf/reference/predict.causal_forest.html, if it is NULL you get out of bag predictions from the trained forest.

**Comment by sudonghua91** [comment]
Hi @erikcs, Thanks for your post. But I did read everything you mentioned above. 

My questions are still there. To have the training sample, do I only need to set a sample.fraction? For instance, it is 0.5. By doing that, half observations form my training set. Then, honesty option keeps the training set being honest, which means half is used for splitting and the other half is used for estimation. Otherwise, I set honety to be FALSE and randomly split sample into half and in turn put the half data into the forest function. Then only sample.fraction is enough to build trees and estimate. 

Prediction: if I put the whole dataset (of course contains training set and test sample) in "newdata", does it cause overfitting? I am asking if predict function is always making out-of-bag predictions regardless of adding the argument "newdata"? Say, when predicting an example i in training set, the algorithms use any trees in which i is never used for either splitting or estimation to make predictions. That is my concern.
Thanks.
